Fix 401 error in share view open file button

andrew-kennedy opened the pull request:
This link would be generated with a `?token=<hash>?inline=true` when it should be `?token=<hash>&inline=true`. This was causing 401 Unauthorized errors because `URL.Query().Get("token")` would return the empty string and so [authenticateShareRequest](https://github.com/filebrowser/filebrowser/blob/dee465ab865ba5bf96fbfbe5888f4e95731d3701/http/public.go#L126) would try to verify the password in the header, which doesn't exist for these requests.

andrew-kennedy committed:
Fix 401 error in share view open file button

This link would be generated with a `?token=<hash>?inline=true` when it should be `?token=<hash>&inline=true`. This was causing 401 Unauthorized errors because `URL.Query().Get("token")` would return the empty string and so (authenticateShareRequest)[https://github.com/filebrowser/filebrowser/blob/dee465ab865ba5bf96fbfbe5888f4e95731d3701/http/public.go#L126] would try to verify the password in the header, which doesn't exist for these requests.

ramiresviana commented on frontend/src/views/Share.vue:
```diff
@@ -93,7 +93,7 @@
             <a
               target="_blank"
-              :href="link + '?inline=true'"
+              :href="link + '&inline=true'"
```
This will make the link invalid on shares without password.

andrew-kennedy replied:
Good point, I really need to properly construct the query string using official builder methods.

andrew-kennedy committed:
Properly handle searchParams

This should avoid the pitfalls of trying to manually decide to attach inline=true with a ? or a &.

andrew-kennedy committed:
Fix missing brace + comma

andrew-kennedy committed:
Update to make linter happy

andrew-kennedy committed:
Merge branch 'master' into patch-1
